### PR TITLE
Run the full test despite failures

### PR DESCRIPTION
**Pull request by siddhesh**
Mark failures and continue execution of the testsuite.  Display a
summary at the end of each compiler run.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,14 +26,23 @@ STATIC_CHECK ?= false
...
 
+RESULTFILE = fortify.result
+
 TARGETS=memcpy memmove mempcpy memset snprintf sprintf stpcpy stpncpy strcat \
 	strcpy strncat strncpy vsnprintf vsprintf
 
 check:$(patsubst %,check-%,$(COMPILERS)) test_common.c
 
+$(RESULTFILE):
+	$(Q)cat /dev/null > $@
```
Comment (serge-sans-paille) [inline Makefile]: What an interesting idiom. I assume it's equivalent to `rm -f $@ && touch $@` ?

File: Makefile [hunk 2]
```diff
@@ -26,14 +26,23 @@ STATIC_CHECK ?= false
...
+
 define check-target =
-check-$(1):$(patsubst %,run_%.$(1),$(TARGETS))
-	@echo "===== $$@ OK ====="
+check-$(1):$(RESULTFILE) $(patsubst %,run_%.$(1),$(TARGETS))
+	$(Q){ echo "===== $$@ DONE ====="; \
+	      echo -n "PASSED: "; grep OK $(RESULTFILE) | wc -l; \
```
Comment (serge-sans-paille) [inline Makefile]: `grep -c` is probably better than `grep | wc -l`

Comment (serge-sans-paille) [reply Makefile]: And `echo -n` => `printf "PASSED"` ?

File: Makefile (hunk 1 above)
Comment (serge-sans-paille) [reply Makefile]: On a more general note, you're using `RESULTFILE` to accumulate the result of checks, which may interact surprisingly if some runs under `make -j`. May I suggest to use a result file per run and `grep` through all of them?

Comment (siddhesh) [reply Makefile]: > What an interesting idiom. I assume it's equivalent to `rm -f $@ && touch $@` ?

Kinda, the `cat` will retain the inode.

Comment (siddhesh) [reply Makefile]: > On a more general note, you're using `RESULTFILE` to accumulate the result of checks, which may interact surprisingly if some runs under `make -j`. May I suggest to use a result file per run and `grep` through all of them?

Good point; I'll split out the results and consolidate in the end.

**Comment by siddhesh** [comment]
Changes:

- Each test writes into its own intermediate test-result file
- At the end of each compiler run, a summary is put into a .result file
- At the end of the full test run, the .result files are dumped, showing the summary of the test run

**Comment by serge-sans-paille** [comment]
LGTM, thanks!

**Pull request merged**